src/DynamoUtilities/DynamoException.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DynamoUtilities
+{
+    /// <summary>
+    /// Represents an error raised by Dynamo
+    /// </summary>
+    [Serializable]
+    public class DynamoException : Exception
```

Comment:
Do we want to be more specific here? I am not sure if this is going to be more widely used, seems that was what you were referring to in the earlier PRs, if that is the case, I would vote for making this one particularly about Python, and we create other DynamoExceptions going forward. But I am actually not sure if @DynamoDS/dynamo team agreed where we are heading. Let's discuss this within team some time

Reply:
The creation of this class was something I saw could bring up a discussion. The problem is this:
- `PythonException` is a Python.NET class. We should not throw these exceptions.
- `Exception` is considered a smell by SQ.
- `ApplicationException` should not be used according to MS.
- We don't have a base exception, that I know of, that says `this error comes from Dynamo`.

This comes from a recommendation I have followed in other projects. You have a base exception like this and if you need specific ones you inherit from this one.

Reply:
Understood. I am supportive but I would like some consensus from team. If we agree, in this case I would be in favor of an inherit `DynamoPythonException` or something similar.

Reply:
I vote with Aaron about making it specific to Python.

Reply:
I vote against, but I think we should probably not have this discussion here. Let's keep this PR open until we can meet to discuss.

Reply:
Are you proposing changing all existing Dynamo exceptions and make them inherit from this one? That doesn't make sense to me.

Reply:
I'm against the proposal of subclassing the base exception because we don't need to handle these exceptions in any particular way. The base type should suffice.